### ISSUE TITLE
Filters and minSpareRows problems at the same time

### ISSUE THREAD

**Issue by dx0eu** [handsontable/handsontable#6278]
### Description

When in hotSettings add filters and minSpareRows.  and when using asynchronous data. the table have a blank row.

### Demo

https://jsfiddle.net/mvtbcwgp/30/

### Your environment
* Handsontable version: 7.1.1
* Browser Name and version: Firefox 65
* Operating System: Linux

**Comment by AMBudnik** [comment]
Thank you for sharing the issue @dx0eu 

I have replicated it with 7.1.1 and tried to update to 8.0.0.-rc1 (beta2)
https://jsfiddle.net/t3rn12sx/ but in the end, there is no data

#### If I used the loadData 
**`(8.0.0.-rc1-beta2)`**
we get 2 empty rows, first and last one
https://jsfiddle.net/t3rn12sx/1/ 
![Zrzut ekranu 2019-09-20 o 14 17 47](https://user-images.githubusercontent.com/10757813/65326311-716a5880-dbb1-11e9-8f76-bea69ac6917c.png)
and here's **`7.1.1.`**
https://jsfiddle.net/Lhnbt3mz/
![Zrzut ekranu 2019-09-20 o 14 21 10](https://user-images.githubusercontent.com/10757813/65326516-f0f82780-dbb1-11e9-96c7-fe0b7433eee6.png)
Both cases are invalid but they differ.

I've tried some older versions on Handsontable and in each case we get behavior from `7.1.1`.

**Comment by wojciechczerniak** [comment]
8.0.0 is the expected output.

1. Handsontable is loaded without data but with `minSpareRows: 1` settings. `dataObj` will be filled with one empty row.
2. Data is pushed to `dataObj` and rows with data are created (note: one empty row is already there).
3. On `loadData` Handsontable checks that last row is not empty and creates one because `minSpareRows: 1` is expected.

You have copied empty row and loaded as data. That's why there are two empty rows now.

Load the data without pushing to `data` object by the reference: https://jsfiddle.net/237fyehd/

@wszymanski WDYT?

**Comment by wojciechczerniak** [comment]
In the example by @dx0eu there is additional `this.getPlugin("Filters").filter()` call.

I think this is very similar to #5208. Maybe a duplicate (@AMBudnik, @aninde WDYT?). My comment from #5208 applies: https://github.com/handsontable/handsontable/issues/5208#issuecomment-533442683 with the small adjustment that the `afterChange` hook source is now `loadData`:

https://jsfiddle.net/uzg391dv/

**Comment by wszymanski** [comment]
> 8.0.0 is the expected output.
> 
> 1. Handsontable is loaded without data but with `minSpareRows: 1` settings. `dataObj` will be filled with one empty row.
> 2. Data is pushed to `dataObj` and rows with data are created (note: one empty row is already there).
> 3. On `loadData` Handsontable checks that last row is not empty and creates one because `minSpareRows: 1` is expected.
> 
> You have copied empty row and loaded as data. That's why there are two empty rows now.
> 
> Load the data without pushing to `data` object by the reference: [jsfiddle.net/237fyehd](https://jsfiddle.net/237fyehd/)
> 
> @wszymanski WDYT?

@wojciechczerniak is right. The behavior is proper and I would suggest using his example.

**Comment by AMBudnik** [comment]
That's true. We had this discussion before. 
A developer should not use any other method for loading data than those recommended in the official API.
I think that we can close the topic.
@dx0eu  I hope that we've answered your question. The latest demo is https://jsfiddle.net/uzg391dv/ (uses `loadData`). If you'd have some additional questions please feel free to let me know.

**Comment by aninde** [comment]
I agree with your statements above, we can close this issue.

**Comment by dx0eu** [comment]
> That's true. We had this discussion before.
> A developer should not use any other method for loading data than those recommended in the official API.
> I think that we can close the topic.
> @dx0eu I hope that we've answered your question. The latest demo is https://jsfiddle.net/uzg391dv/ (uses `loadData`). If you'd have some additional questions please feel free to let me know.


Ok thanks. `loadData` is good. but when is use handsontable in vue. how to do call this function. I wrote a case that is more in line with the current situation.

https://jsfiddle.net/2zbsruva/4/

**Comment by AMBudnik** [comment]
Treating the array data as a simple array and using the `push`/`pop` methods won't work for Handsontable. Use of the official API (`data` and `loadData`) are only methods that can guarantee that the data will load correctly. It may look like the same action but underneath it triggers data processing that may fail or never execute while using `push()`.

We're aware that it would be great to be able to render only the rows that have changed, as currently, you need to load a whole new table. That can slow down the performance. However, this will change in the future. This topic is further explained here https://github.com/handsontable/handsontable/issues/5769

**Comment by dx0eu** [comment]
In the last demo    https://jsfiddle.net/2zbsruva/8/    , I `push` to `pageData` not `push` to `hotSettings.data`  and finally call `hot.loadData(pageData)`  , Not the expected way of working. So push to `pageData` is not possible?

I wrote another test. not use push function,  https://jsfiddle.net/yLmquza5/3/

```javascript
vm.$refs.hot.hotInstance.loadData(Handsontable.helper.createSpreadsheetData(6, 10));
```

It does't work, The second row is also empty.

**Comment by dx0eu** [comment]
I wrote some examples.

|   | Work  | Demo  |
|---|---|---|
| JavaScript  | Yes  | https://jsfiddle.net/vjtbph2k/  |
| Vue  | No | https://jsfiddle.net/yLmquza5/9/  |
|  React | No  | https://jsfiddle.net/4z7y1twh/6/  |

**Comment by AMBudnik** [comment]
Thank you for doing the investigation. I will help us a lot with further tests.

**Comment by wszymanski** [comment]
@dx0eu, thanks for your work. In examples for `Vue` and `React` you loaded scripts for `7.1.1`. Please try our last build:

https://cdn.jsdelivr.net/gh/handsontable/handsontable@db069ba1ebea815da0375f69244361cfe13eab98/dist/handsontable.full.js
https://cdn.jsdelivr.net/gh/handsontable/handsontable@db069ba1ebea815da0375f69244361cfe13eab98/dist/handsontable.full.css

If I understand you correctly it will work. Please let us know whether everything is right.

**Comment by dx0eu** [comment]
Yes, that's it. Looking forward to the next version

**Comment by wojciechczerniak** [comment]
Please keep it open for now. We will close the issue after `8.0.0` release.

**Comment by AMBudnik** [comment]
Hi @dx0eu 

We've just released version 8.0.0-beta1 where this issue is fixed https://jsfiddle.net/gh9pc8o3/
As this is only a beta now we do not recommend using it on the production. However, we wanted to share the version for you to test so any comment highly appreciated 👍

**Comment by AMBudnik** [comment]
Still works well in 8.0.0-beta2-rev6 https://jsfiddle.net/AMBudnik/oje8ct56/

**Comment by AMBudnik** [comment]
Hi @dx0eu I'm happy to inform you that this issue is officially fixed in v 8.0.0 https://jsfiddle.net/m5by6wrn/
Thank you for your patience. I hope that the amount of work that we put on the v8 will help you to improve your project.

CDN https://cdn.jsdelivr.net/gh/handsontable/handsontable@latest/dist/handsontable.full.css
https://cdn.jsdelivr.net/gh/handsontable/handsontable@latest/dist/handsontable.full.js
NPM npm i handsontable
Release Notes https://github.com/handsontable/handsontable/releases  

I highly encourage you to test the new version and leave some feedback.
